src/foundry/foundry.js/clientDocuments/chatMessage.d.ts

```diff
@@ -71,10 +71,10 @@ declare global {
      * @param rollMode - The rollMode preference to apply to this message data
      * @returns The modified ChatMessage data with rollMode preferences applied
      */
-    static applyRollMode(
-      chatData: foundry.data.ChatMessageData,
+    static applyRollMode<Data extends foundry.data.ChatMessageData>(
+      chatData: Data,
```

Comment:
I'm confused, didn't you want to make it _not_ static and use the constructor data instead? I.e.
```suggestion
    static applyRollMode(
      chatData: ConstructorDataType<foundry.data.ChatMessageData>,
```
(`ConstructorDataType` is a helper type to get the constructor data type from some class extending `DocuentData`)

Reply:
I'm stupid. Left changes in my index, then dipped off to go back to my system.

As not making it *not* static, that was never the point. The point was it was static and I could prep my data before creating the message proper.

Reply:
Sorry, I misspelled, I wanted to say _not_ generic 🤦‍♂️